docs/wiki/Cryptography/bls.md

```diff
@@ -0,0 +1,232 @@
+
+BLS signatures, employing bilinear pairings, offer robust protection against certain cryptographic attacks and produce shorter signatures. Unlike Schnorr, BLS does not rely on random number generation for securing signatures, making it inherently more secure against randomness-related vulnerabilities.
+
+#### Example of BLS Signature Generation and Verification:
+
+<figure class="diagram" style="width:80%">
+
+![Diagram showing key pair generation and verification for BLS](../../images/elliptic-curves/bls-alice.png)
+
+<figcaption>
+
+_Visual Aid to understand how BLS signatures work_
+
+</figcaption>
+</figure>
+
+Consider Alice creating a BLS signature. She starts with her private key $a$, and computes her public key $P$ using a generator point $G$ on the elliptic curve:
+
+$$ P = aG $$
+
+She hashes her message and maps this hash to a point on the curve, $H(M)$. Her signature $S$ is then:
+
+$$ S = a \times H(M) $$
+
+The signature is verified using the pairing function:
+
+$$ e(G, S) = e(P, H(M)) $$
+
+This can be proven as: $$e(G,S)=e(G,a×H(m))=e(a×G,H(m))=e(P,H(M))$$ where $G$ is the generator point on the elliptic curve.
+
+This equation proves that the signature was indeed created by the holder of the private key corresponding to $P$.
+
+#### Example
+
+For BLS signatures using a curve like BLS12-381 example values would look like:
+
+```
+Message: "Hello"
+Secret Key: 26daf744780a51072aa8de191259bf7ff080b8457512cfd0eedfb4f8c71b131d
+Public Key: bfdab807246849b76b7bdf5229619b9ccb33713633644a48b7ab3a7e67af7c1ae9d597a1c0fac6f61e63c1278b26c2f527be3d58bce95451b36f0c692ee90e1f
+Signature: dee15784b458419b4b8bbdbb13838da13c27dccab6ef50f0dcb4ff7352048c0b
+```
+
+For ECDSA using a curve like secp256k1, there's an $R$ and an $S$ value, which produces a longer signature whose example values would look like:
+
+```
+Message: "Hello"
+Private Key: 2aabe11b7f965e8b16f525127efa01833f12ccd84daf9748373b66838520cdb7
+Public Key (EC Point):
+    x: 39516301f4c81c21fbbc97ada61dee8d764c155449967fa6b02655a4664d1562
+    y: d9aa170e4ec9da8239bd0c151bf3e23c285ebe5187cee544a3ab0f9650af1aa6
+Signature:
+    R: 905eceb65a8de60f092ffb6002c829454e8e16f3d83fa7dcd52f8eb21e55332b
+    S: 8f22e3d95beb05517a1590b1c5af4b2eaabf8e231a799300fffa08208d8f4625
+```
+
+### Aggregation in BLS:
+
+A major advantage of BLS is the ability to aggregate multiple signatures into a single compact signature. This is particularly useful in scenarios involving multiple transactions or signers, greatly reducing the blockchain space and computational power needed for verifications. For example if there are 100 transactions, where signature for each one is represented by $S_i$ and each are associated with a public key of $P_i$ (and a message $M_i$), rather than storing 100 separate signatures, BLS allows combining them into one:
+
+$$ S = S*1 + S_2 + \ldots + S*{100} $$
```

Comment:
Not rendering correctly in GitHub. It should be:

$$S = S_1 + S_2 + \ldots + S_{100}$$